stan/math/prim/scal/prob/beta_binomial_lpmf.hpp

```diff
@@ -56,68 +55,96 @@ return_type_t<T_size1, T_size2> beta_binomial_lpmf(const T_n& n, const T_N& N,
                          "First prior sample size parameter", alpha,
                          "Second prior sample size parameter", beta);
 
-  if (!include_summand<propto, T_size1, T_size2>::value) {
+  if (!include_summand<propto, T_size1, T_size2>::value)
     return 0.0;
-  }
 
   operands_and_partials<T_size1, T_size2> ops_partials(alpha, beta);
 
-  const scalar_seq_view<T_n> n_vec(n);
-  const scalar_seq_view<T_N> N_vec(N);
-  const scalar_seq_view<T_size1> alpha_vec(alpha);
-  const scalar_seq_view<T_size2> beta_vec(beta);
-  const auto size = max_size(n, N, alpha, beta);
+  scalar_seq_view<T_n> n_vec(n);
+  scalar_seq_view<T_N> N_vec(N);
+  scalar_seq_view<T_size1> alpha_vec(alpha);
+  scalar_seq_view<T_size2> beta_vec(beta);
+  size_t size = max_size(n, N, alpha, beta);
 
   for (size_t i = 0; i < size; i++) {
     if (n_vec[i] < 0 || n_vec[i] > N_vec[i]) {
       return ops_partials.build(LOG_ZERO);
     }
   }
 
-  if (include_summand<propto>::value) {
-    for (size_t i = 0; i < size; i++) {
-      // normalizing constant
-      logp += binomial_coefficient_log(N_vec[i], n_vec[i]);
-      // log numerator - denominator
-      logp += lbeta(n_vec[i] + value_of(alpha_vec[i]),
-                    N_vec[i] - n_vec[i] + value_of(beta_vec[i]))
-              - lbeta(value_of(alpha_vec[i]), value_of(beta_vec[i]));
-    }
-  }
-
-  VectorBuilder<!is_constant_all<T_size1, T_size2>::value, T_partials, T_N,
-                T_size1, T_size2>
-      digamma_N_plus_alpha_plus_beta(size);
-  VectorBuilder<!is_constant_all<T_size1, T_size2>::value, T_partials, T_size1,
-                T_size2>
-      digamma_alpha_plus_beta(size);
-  if (!is_constant_all<T_size1, T_size2>::value) {
-    for (size_t i = 0; i < size; i++) {
+  VectorBuilder<include_summand<propto>::value, T_partials_return, T_n, T_N>
+      normalizing_constant(max_size(N, n));
+  for (size_t i = 0; i < max_size(N, n); i++)
+    if (include_summand<propto>::value)
+      normalizing_constant[i] = binomial_coefficient_log(N_vec[i], n_vec[i]);
+
+  VectorBuilder<include_summand<propto, T_size1, T_size2>::value,
+                T_partials_return, T_n, T_N, T_size1, T_size2>
+      lbeta_numerator(size);
+  for (size_t i = 0; i < size; i++)
+    if (include_summand<propto, T_size1, T_size2>::value)
```

Comment:
This line and lines 93 and 135 could be removed because in line 58 we have
```
if (!include_summand<propto, T_size1, T_size2>::value)
    return 0.0;	    return 0.0;
```
which means that if this isnt true we never reach this anyways. This was cleaned in #1520 by @mcol. 

These all get compiled out so there is no performance penalty and I would rather fix the bug first and then clean this in subsequent PR.